### PR TITLE
docs: added clarity in lifecycle event docs

### LINKED ISSUE #49686: Fully initialisation meaning clarification in ngAfterContentInit
### Describe the problem that you experienced

Unable to understand meaning of ngAfterContentInit and ngAfterContentChecked life-cycle hook description.

### Enter the URL of the topic with the problem

https://angular.io/api/core/AfterContentInit 
https://angular.io/api/core/AfterContentChecked

### Describe what you were looking for in the documentation

What does "fully initialised"  mean in AfterContentinit and "has completed checking all content" in AfterContentChecked? 

### Describe the actions that led you to experience the problem

I have fully described the issue and experiment I did in [stackoverflow](https://stackoverflow.com/questions/75914528/what-is-the-correct-contract-for-ngaftercontentinit-and-ngaftercontentchecked)
Please check link for full description.

### Describe what you want to experience that would fix the problem

Clarify the docs with profound example.

### PR DESCRIPTION
**Pull request by sumitparakh**
Closes #49686

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by tokidev-ai — APPROVED** [review]
👍🏻

File: packages/core/src/interface/lifecycle_hooks.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export interface OnDestroy {
 /**
  * @description
  * A lifecycle hook that is called after Angular has fully initialized
- * all content of a directive.
+ * all content of a directive. It wil run only once when the projected content is initialized.
```
Comment (mmoroteh) [inline packages/core/src/interface/lifecycle_hooks.ts:130]: Maybe typo

Comment (sumitparakh) [reply packages/core/src/interface/lifecycle_hooks.ts:130]: Hi @mmoroteh, maybe I didn't understand your comment. What typo mistake you are referring to?

Comment (mmoroteh) [reply packages/core/src/interface/lifecycle_hooks.ts:130]: imagen

Comment (sumitparakh) [reply packages/core/src/interface/lifecycle_hooks.ts]: Aah. How did I miss it. Thanks. Fix it.

**Comment by JeanMeche** [comment]
Hi, thank you for improving the docs. 

For your PR to move forward, can you squash your commits and instead of merging main, rebase on it.

**Comment by sumitparakh** [comment]
I think I messed up my git tree.

**Comment by sumitparakh** [comment]
> Hi, thank you for improving the docs.
> 
> For your PR to move forward, can you squash your commits and instead of merging main, rebase on it.

Done

**Review by jessicajaniuk — APPROVED** [review]
LGTM. Thank you.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 2f1796f8f6299d78de7e286e5b836d8402bb7661.

**Pull request closed without merging**